src/renderers/shared/stack/reconciler/ReactRef.js

```diff
@@ -56,20 +66,21 @@ ReactRef.shouldUpdateRefs = function(prevElement, nextElement) {
   // is made. It probably belongs where the key checking and
   // instantiateReactComponent is done.
 
-  var prevEmpty = prevElement === null || prevElement === false;
-  var nextEmpty = nextElement === null || nextElement === false;
-
   return (
     // This has a few false positives w/r/t empty components.
-    prevEmpty || nextEmpty ||
+    prevElement == null || prevElement === false || // prevEmpty
```

Comment:
It might make sense to flowify from the other angle then if we can't figure this out.

Based on the comment above, it seems like potentially the case where we're swapping refs from something that was rendered to something that isn't, so the instance doesn't have a prevElement (or something).